QOL Artifact changes

MuckerMayhem opened the pull request:
🆑 Mucker
rscdel: Removes gas generation artifact effects.
bugfix: The artifact scanner will now properly detect secondary artifact effects.
bugfix: Fixed the 'Touch' trigger for artifacts, poking rocks now actually has consequences!
tweak: Artifact effects will now remain on for longer when activated in order for effects to have time to happen.
tweak: 'Pushback' artifact effect will no longer tear off limbs.
/🆑

Gas generation effects were removed as they were painfully boring and uninteresting. Also slightly changes how artifacts are turned on, so that they stay on for 5 - 20 second intervals, to allow for the effects to actually happen.

SierraKomodo commented on code/modules/xenoarcheaology/effect.dm:
```diff
@@ -115,6 +124,10 @@
 
+/datum/artifact_effect/proc/toggle_off()
+	toggled = FALSE
+	ToggleActivate(1)
```
```suggestion
	ToggleActivate(TRUE)
```

SierraKomodo commented on code/modules/admin/admin.dm:
```diff
@@ -1147,7 +1147,7 @@ var/global/floorIsLava = 0
 			A.secondary_effect.trigger = new secondary_trigger
 		else
-			A.secondary_effect = null
+			QDEL_NULL(A.secondary_effect)
```
QDEL seems unnecessary here - `A` is a new instance at this point so `secondary_effect` shouldn't actually exist.

MuckerMayhem replied:
The parent's `new()` creates a secondary status effect that just setting to null didn't seem to stop from existing.

SierraKomodo commented on code/modules/xenoarcheaology/artifacts/artifact.dm:
```diff
@@ -8,6 +8,7 @@
 	var/datum/artifact_effect/my_effect
 	var/datum/artifact_effect/secondary_effect
 	var/being_used = 0
+	var/on_time = 100
```
Somewhat redundant and misleading to have a default value here when it's reassigned on `New()`

SierraKomodo commented on code/modules/xenoarcheaology/effect.dm:
```diff
@@ -3,11 +3,13 @@
 	var/artifact_id = ""
 	var/effect_type = 0
+	var/toggled = FALSE
+	var/on_time = 100 //time artifact should stay on for when toggled
```
Same as above - Moreso redundant because this seems to be the exact same as the artifact's `on_time` var. Does this need to exist across both paths for a specific reason?

MuckerMayhem replied:
One of those is a leftover. whoops!!!

SierraKomodo reviewed (changes requested):
Also, spawning an artifact causes a runtime:

```
[14:01:53] Runtime in code/controllers/subsystems/timer.dm,464: addtimer called with a callback assigned to a qdeleted object. In the future such timers will not be supported and may refuse to run or run with a 0 wait
  proc name: crash at (/proc/crash_at)
  usr: Zhai Kovac (/mob/observer/ghost) ([0x3000090]) (the deck) (84,97,3) (/turf/simulated/floor/tiled/white) (sierrakomodo)
  usr.loc: The deck (/turf/simulated/floor/tiled/white) ([0x10183d3]) (84,97,3) (/area/crew_quarters/sleep/cryo)
  src: null
  call stack:
  crash at("addtimer called with a callbac...", "code/controllers/subsystems/ti...", 464)
  addtimer(/datum/callback (/datum/callback), 160, 0)
  good feeling (/datum/artifact_effect/goodfeeling): ToggleActivate(0)
```

MuckerMayhem commented:
Removed a case of `spawn` and used a timer instead, which fixes the runtime. Also discovered that the touch effect for artifacts never actually worked, and added a case to make those work (could be better, but frankly the whole of artifacts needs a bit of a facelift)

MuckerMayhem committed:
artifact updates and fixes

Tweaks pushback so it doesn't tear off limbs, fixed the analyzer not detecting secondary effects, removed gas effects, added toggle to artifact effects when triggered so they stay on longer.
